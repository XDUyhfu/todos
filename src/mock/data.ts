import dayjs from "dayjs";

export enum Status {
  DONE = "已完成",
  UNDONE = "未完成",
}

export interface TodoItem {
  title: string;
  content: string;
  createTime: string;
  status: Status;
}

export const GenerateUndoItem = () => ({
  title: "这是一个测试的todo",
  content: "我今天要完成这件事情",
  createTime: dayjs().format("YYYY-MM-DD HH:mm:ss"),
  status: Status.UNDONE,
});

export const data: TodoItem[] = [
  {
    title: "这是一个测试的todo1",
    content: "我今天要完成这件事情",
    createTime: dayjs().format("YYYY-MM-DD HH:mm:ss"),
    status: Status.DONE,
  },
  {
    title: "这是一个测试的todo",
    content: "我今天要完成这件事情",
    createTime: dayjs().format("YYYY-MM-DD HH:mm:ss"),
    status: Status.DONE,
  },
  {
    title: "这是一个测试的todo",
    content: "我今天要完成这件事情",
    createTime: dayjs().format("YYYY-MM-DD HH:mm:ss"),
    status: Status.UNDONE,
  },
  {
    title: "这是一个测试的todo",
    content: "我今天要完成这件事情",
    createTime: dayjs().format("YYYY-MM-DD HH:mm:ss"),
    status: Status.UNDONE,
  },
  {
    title: "这是一个测试的todo",
    content: "我今天要完成这件事情",
    createTime: dayjs().format("YYYY-MM-DD HH:mm:ss"),
    status: Status.DONE,
  },
];
