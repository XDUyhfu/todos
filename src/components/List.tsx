import { TodoItem } from "../mock/data.ts";

export const List = ({
  data,
  deleteItem,
}: {
  data: TodoItem[];
  deleteItem: (item: TodoItem) => void;
}) => {
  return (
    <div>
      {data.map((item) => {
        return (
          <div
            key={item.key}
            role="list-item"
            style={{ border: "1px solid #ccc", marginBottom: 10 }}
          >
            标题:
            <span>{item.title}</span>
            <br />
            内容: <span>{item.content}</span>
            <br />
            创建时间: <span>{item.createTime}</span>
            <br />
            状态:
            <span>{item.status}</span>
            <br />
            <button onClick={() => deleteItem(item)}>删除</button>
          </div>
        );
      })}
    </div>
  );
};
