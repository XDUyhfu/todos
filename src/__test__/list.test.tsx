import { beforeEach, describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { List } from "../components/List.tsx";
import { data, Status } from "../mock/data.ts";

describe("List组件测试", () => {
  beforeEach(() => {
    render(<List data={data} />);
  });

  it("渲染数量", () => {
    const elements = screen.getAllByRole("list-item");
    expect(elements.length).toBe(data.length);
  });

  it("已完成数量", () => {
    const elements = screen.getAllByText(Status.DONE);
    expect(elements.length).toBe(
      data.filter((item) => item.status === Status.DONE).length,
    );
  });

  it("未完成数量", () => {
    const elements = screen.getAllByText(Status.UNDONE);
    expect(elements.length).toBe(
      data.filter((item) => item.status === Status.UNDONE).length,
    );
  });

  it("快照测试", () => {
    const elements = screen.getAllByRole("list-item");
    expect(elements).toMatchSnapshot();
  });
});
