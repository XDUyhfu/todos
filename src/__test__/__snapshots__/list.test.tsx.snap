// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`List组件测试 > 快照测试 1`] = `
[
  <div
    role="list-item"
    style="border: 1px solid #ccc; margin-bottom: 10px;"
  >
    标题:
    <span>
      这是一个测试的todo1
    </span>
    <br />
    内容: 
    <span>
      我今天要完成这件事情
    </span>
    <br />
    创建时间: 
    <span>
      2024-02-03 22:14:39
    </span>
    <br />
    状态:
    <span>
      已完成
    </span>
  </div>,
  <div
    role="list-item"
    style="border: 1px solid #ccc; margin-bottom: 10px;"
  >
    标题:
    <span>
      这是一个测试的todo
    </span>
    <br />
    内容: 
    <span>
      我今天要完成这件事情
    </span>
    <br />
    创建时间: 
    <span>
      2024-02-03 22:14:39
    </span>
    <br />
    状态:
    <span>
      已完成
    </span>
  </div>,
  <div
    role="list-item"
    style="border: 1px solid #ccc; margin-bottom: 10px;"
  >
    标题:
    <span>
      这是一个测试的todo
    </span>
    <br />
    内容: 
    <span>
      我今天要完成这件事情
    </span>
    <br />
    创建时间: 
    <span>
      2024-02-03 22:14:39
    </span>
    <br />
    状态:
    <span>
      未完成
    </span>
  </div>,
  <div
    role="list-item"
    style="border: 1px solid #ccc; margin-bottom: 10px;"
  >
    标题:
    <span>
      这是一个测试的todo
    </span>
    <br />
    内容: 
    <span>
      我今天要完成这件事情
    </span>
    <br />
    创建时间: 
    <span>
      2024-02-03 22:14:39
    </span>
    <br />
    状态:
    <span>
      未完成
    </span>
  </div>,
  <div
    role="list-item"
    style="border: 1px solid #ccc; margin-bottom: 10px;"
  >
    标题:
    <span>
      这是一个测试的todo
    </span>
    <br />
    内容: 
    <span>
      我今天要完成这件事情
    </span>
    <br />
    创建时间: 
    <span>
      2024-02-03 22:14:39
    </span>
    <br />
    状态:
    <span>
      已完成
    </span>
  </div>,
]
`;
